x-pack/legacy/plugins/maps/public/shared/components/metric_editor.js

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { Fragment } from 'react';
+import PropTypes from 'prop-types';
+import { i18n } from '@kbn/i18n';
+
+import {
+  EuiFieldText,
+  EuiFormRow,
+} from '@elastic/eui';
+
+import {
+  MetricSelect,
+  METRIC_AGGREGATION_VALUES,
+} from './metric_select';
+import { SingleFieldSelect } from './single_field_select';
+
+export function MetricEditor({ fields, metricsFilter, metric, onChange }) {
+
+  const onAggChange = (metricAggregationType) => {
+    onChange({
+      ...metric,
+      type: metricAggregationType,
+    });
+  };
+  const onFieldChange = (fieldName) => {
+    onChange({
+      ...metric,
+      field: fieldName,
+    });
+  };
+  const onLabelChange = (e) => {
```

Comment:
this triggers a requerying of the data, which we should avoid.

Maybe in the `VectorLayer#_syncJoin`, we should check if the actual join-config has changed.